env0/utils.go

```diff
@@ -41,31 +41,28 @@ func intPtr(i int) *int {
 	return &i
 }
 
-// Extracts values from the resourcedata, and writes it to the interface.
-func readResourceData(i interface{}, d *schema.ResourceData) error {
+// Prepends prefix to the fieldName.
+func readResourceDataEx(prefix string, i interface{}, d *schema.ResourceData) error {
 	// TODO: add a mechanism that returns an error if fields were set in the resourceData but not in the struct.
 	// Blocked by: https://github.com/hashicorp/terraform-plugin-sdk/issues/910
 
 	val := reflect.ValueOf(i).Elem()
 	for i := 0; i < val.NumField(); i++ {
-		fieldName := val.Type().Field(i).Name
-		// Assumes golang is CamalCase and Terraform is snake_case.
-		// This behavior can be overrided be used in the 'tfschema' tag.
-		fieldNameSC := toSnakeCase(fieldName)
-		if resFieldName, ok := val.Type().Field(i).Tag.Lookup("tfschema"); ok {
-			if resFieldName == "-" {
-				continue
-			}
+		parsedField := getFieldName(val.Type().Field(i))
```

Comment:
Question:
Why here we get the field with
```val.Type().Field(i)```

While in line 61, we get the field with:
```field := val.Field(i)```

Is there a need for `.Type()`?

Reply:
I would think so (but yes agree it's confusing).
We don't want the field itself, but the metadata of the field.
I believe val.Type() returns the struct meta data.
With the metadata you can extract the field name.

In line 61 we're more interested in the field value.

https://pkg.go.dev/reflect#Type

I'll go over the code down the line and see if it can be done in a better way.